src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const state2 = { ...state };
+  const log = [];
+
+  for (const action in actions) {
+    switch (actions[action].type) {
```

Comment:
would be nice to create a variable for `actions[action]` - you're using it often. It will increase readability (especially in such cases as in the 25 row)

Reply:
done )